modules/runner/src/integrationTest/java/org/apache/ignite/internal/raftsnapshot/ItTableRaftSnapshotsTest.java

```diff
@@ -792,6 +816,116 @@ void testChangeLeaderDuringSnapshotInstallationToLeaderWithEnoughLog() throws Ex
         assertThat(rows, is(List.of(new IgniteBiTuple<>(1, "one"))));
     }
 
+    /**
+     * The replication mechanism must not replicate commands for which schemas are not yet available on the node
+     * to which replication happens (in Raft, it means that followers/learners cannot receive commands that they
+     * cannot execute without waiting for schemas). This method tests that snapshots bringing such commands are
+     * rejected, and that, when metadata catches up, the snapshot gets successfully installed.
+     */
+    @Test
+    void laggingSchemasOnFollowerPreventSnapshotInstallation() throws Exception {
+        cluster.startAndInit(3);
+
+        createTestTableWith3Replicas(DEFAULT_STORAGE_ENGINE);
+
+        // Prepare the scene: force node 0 to be a leader, and node 2 to be a follower.
+
+        final int leaderIndex = 0;
+        final int followerIndex = 2;
+
+        transferLeadershipOnSolePartitionTo(leaderIndex);
+        cluster.transferLeadershipTo(leaderIndex, MetastorageGroupId.INSTANCE);
+
+        // Block AppendEntries from being accepted on the follower so that the leader will have to use a snapshot.
+        blockIncomingAppendEntriesAt(followerIndex);
```

Comment:
We could create a table on nodes `A` and `B`, and then increase the number of replicas, causing the rebalance process and creation of the table on node `C`. If we truncate the log before that, it would lead to the full state transfer. No need to block append entries requests, but it'll probably get harder to inhibit the meta-storage...
Anyway, that wasn't a suggestion, it was a conversation starter

Reply:
The problem is that if node C's watches are inhibited it will not see the assignments update, so the rebalance will not happen.

Also, using regular mechanisms in such a way looks as a hack itself. It seems that when we want to cripple the ability of a node to receive AppendEntries we should do exactly this, it makes the test a little bit understandable and less fragile.